vulcanai2/datasets/tabulardataset.py

```diff
@@ -15,54 +15,87 @@
 
 class TabularDataset(Dataset):
     """
-    This defines a dataset, subclassed from torch.utils.data.Dataset. It uses pd.dataframe as the backend, with utility
+    This defines a dataset, subclassed from torch.utils.data.Dataset.
+    It uses pd.dataframe as the backend, with utility
     functions.
     """
-    #TODO: rewrite to work with new stitch datasets function
-    def __init__(self, data, label_column, join_column=None, index_list=None, na_values=None):
-        """
-        Creates an instance of Tabulardataset
-        :param data: Either a path to a csv file, a list of paths to csv files, a list of dataframes, or a dataframe.
-        :param label_column: Must specify, provide None if you do not want a target. Must be unique.
-        :param join_column: The column on which a list of datasets should be joined
-        :param index_list: list of feature columns to index on when stitching(default None)
-        :return: None
-        """
-        if isinstance(data, list):
-            if not join_column:
-                raise RuntimeError("You need to provide a join_column if a list of csvs or dataframes are provided")
-            if isinstance(data[0], str):
-                dfs = [pd.read_csv(f, na_values=na_values) for f in data]
-            self.df = utils.stitch_datasets(dfs, join_column, index_list)
-        elif isinstance(data, pd.DataFrame):
-            self.df = data
+    def __init__(self, label_column=None, merge_on_columns=None,
+                 index_list=None, na_values=None, **kwargs):
+        """
+        Creates an instance of TabularDataset
+
+        Parameters
+        ----------
+        label_column: String
+            The name of the label column.
+            Provide None if you do not want a target.
+        merge_on_columns: list of strings
+            Key(s) that specifies which columns to use to uniquely stitch
+            dataset (default None)
+        index_list: list of strings
+            List of columns to make the index of the dataframe
+        na_values: The values to convert to NaN when reading from csv
+        kwargs: keyword parameter, value is either path or dataframe
+            Where key: dataset name and value is either a path to a file
+            or a dataframe.
+        """
+        dataset_dict = kwargs
```

Comment:
if it's the dataset_dict, shouldn't we call the function parameter that instead of `kwargs`?

Reply:
... that is an excellent point. I kinda just, defaulted to kwargs

Reply:
kwargs is normally left for other case-by-case parameters that would need to be passed